src/NServiceBus.RavenDB.Tests/Timeouts/When_adding_timeouts.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace NServiceBus.RavenDB.Tests.Timeouts
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Extensibility;
+    using NServiceBus.TimeoutPersisters.RavenDB;
+    using NUnit.Framework;
+    using TimeoutData = NServiceBus.Timeout.Core.TimeoutData;
+
+    public class When_adding_timeouts : RavenDBPersistenceTestBase
+    {
+        [Test]
+        public async Task Add_WhenNoIdProvided_ShouldSetDbGeneratedTimeoutId()
```

Comment:
The idea going forward is to pass in a stable "timeout id" to allow the storage to deduplicate.

Perhaps a `TryAdd(timeOutData, out nativeId)` ?

That said the core isn't using that native id so if we passed it to the add, `Add(timeoutId, timeoutData)` we wouldn't need to return it since it's already known by the caller?